### ISSUE TITLE
Install the RATS-TLS library in compile env to fix dependency bugs.

### ISSUE THREAD

**Issue by HaokunX-intel** [confidential-containers/enclave-cc#76]
Although the PR #48 installs RATS-TLS in the runtime env, if we want to build a enclave agent referring to latest image-rs, which depends on the RATS-TLS, we need to install the RATS-TLS in the compile env too.

Otherwise, the ld complains that it can not find `rats-tls` as shown in [here](https://github.com/confidential-containers/enclave-cc/actions/runs/3683353586/jobs/6231836441)
```
 = note: /usr/bin/ld: cannot find -lrats_tls
          collect2: error: ld returned 1 exit status
```

The building [Dockerfile](https://github.com/confidential-containers/enclave-cc/blob/main/tools/packaging/build/agent-enclave-bundle/Dockerfile) should install the rats-tls in the builder, it may look like
```
FROM rust:1.63-bullseye as builder


RUN apt-get update && \
    env DEBIAN_FRONTEND=noninteractive apt-get install -y \
    protobuf-compiler

# FIX: install rats-tls
RUN git clone --depth 1 https://github.com/inclavare-containers/rats-tls.git && \ 
    cd rats-tls && \
    cmake -DRATS_TLS_BUILD_MODE="occlum" -DBUILD_SAMPLES=on -H. -Bbuild && \
    make -C build install

# Build enclave-agent
COPY src/ /enclave-cc/src/
RUN cd /enclave-cc/src/enclave-agent && \
    rustup component add rustfmt && \
    make

# Start preparing enclave-agent "bundle"
FROM ubuntu:20.04
...

```

**Comment by mythi** [comment]
Would it be possible to get `rats-tls` installable from their Deb repo? http://mirrors.openanolis.cn/inclavare-containers/deb-repo/pool/main/r/rats-tls/

**Comment by haosanzi** [comment]
hi, rats-tls can provide deb package. Put it in the repo of ic. 

just FYI: rats-tls will be updated recently to support the DICE standard. The relevant pr is at https://github.com/inclavare-containers/rats-tls/pull/123

**Comment by HaokunX-intel** [comment]
I agree with @haosanzi . My concern is updating official repo may bring more overhead, which is too heavy to us currently. Given that rats-tls and enclave-cc update rapidly, we need more flexible paths to combine them. I think we can build the rats-tls with certain commit id  from source code to avoid compatible problems.

Once the enclave-cc is planned to release for commercial use or the rats-tls becomes stable, we can install rats-tls from the official repo.

**Comment by mythi** [comment]
Building is fine but: let's not do it twice, let's build from known revisions instead of repo's `HEAD` which is a moving target and can trigger build errors anytime.

**Comment by HaokunX-intel** [comment]
Closed by PR #77
